src/build_context.rs

```diff
@@ -153,60 +124,12 @@ impl BuildContext<CombinedValidator> {
 impl BuildContext<CombinedSerializer> {
     /// Move validators into a new vec which maintains the order of slots
     pub fn into_slots_ser(self) -> PyResult<Vec<CombinedSerializer>> {
-        self.slots
+        self.definitions
             .into_iter()
-            .map(|slot| {
-                slot.op_val_ser
-                    .ok_or_else(|| py_error_type!("Slots Error: slot not yet filled"))
+            .map(|slot| match slot {
+                None => py_err!("Slots Error: slot not yet filled"),
+                Some(v) => Ok(v),
             })
             .collect()
     }
 }
-
-fn extract_used_refs(schema: &PyAny, refs: &mut AHashSet<String>) -> PyResult<()> {
-    if let Ok(dict) = schema.downcast::<PyDict>() {
-        if is_definition_ref(dict)? {
-            refs.insert(dict.get_as_req(intern!(schema.py(), "schema_ref"))?);
-        } else {
-            for (key, value) in dict.iter() {
-                if !key.eq(intern!(schema.py(), "metadata"))? {
-                    extract_used_refs(value, refs)?;
-                }
-            }
-        }
-    } else if let Ok(list) = schema.downcast::<PyList>() {
-        for item in list.iter() {
-            extract_used_refs(item, refs)?;
-        }
-    }
-    Ok(())
-}
```

Comment:
It seems like a lot of the dragons were in here, and it was slow (hence the building schema speedup).

Reply:
makes sense.

Reply:
i guess this is how we did the reference counting.